File: Okusuri_nometane/fix_style.js
const fs = require('fs');
const path = require('path');

const filePath = path.join('c:', 'Users', 'kiyoshi', 'Github_repository', 'search-for-medicine', 'Okusuri_nometane', 'style.css');

try {
    let content = fs.readFileSync(filePath, 'utf8');
    let lines = content.split('\n');

    // 1. Fix mobile layout
    let targetStartIndex = -1;
    for (let i = 0; i < lines.length; i++) {
        if (i > 1500 && lines[i].includes('.certificate-characters {')) {
            targetStartIndex = i;
            break;
        }
    }

    if (targetStartIndex !== -1) {
        if (lines[targetStartIndex + 1].includes('column')) {
            const newContent = [
                '    .certificate-characters {',
                '        flex-direction: row;',
                '        flex-wrap: nowrap;',
                '        gap: 5px;',
                '        justify-content: center;',
                '    }',
                '',
                '    .character-signature img {',
                '        width: 50px;',
                '        height: 50px;',
                '    }',
                '',
                '    .character-signature span {',
                '        font-size: 0.8rem;',
                '    }'
            ];
            // Replace 4 lines
            lines.splice(targetStartIndex, 4, ...newContent);
            console.log('Mobile layout fixed.');
        } else {
            console.log(`Target content mismatch at line ${targetStartIndex + 1}: ${lines[targetStartIndex + 1]}`);
        }
    } else {
        console.log('Could not find .certificate-characters block.');
    }

    // 2. Remove duplicates
    // Recalculate lines because we modified the array
    // Find start of duplicate block
    let dupStartIndex = -1;
    for (let i = 0; i < lines.length; i++) {
        if (i > 1570 && lines[i].includes('.certificate-body {')) {
            dupStartIndex = i;
            break;
        }
    }

    let dupEndIndex = -1;
    if (dupStartIndex !== -1) {
        for (let i = dupStartIndex; i < lines.length; i++) {
            if (lines[i].includes('.certificate-characters {')) {
                if (i + 1 < lines.length && lines[i + 1].includes('column')) {
                    dupEndIndex = i + 4; // Closing brace of media query
                    break;
                }
            }
        }
    }

    if (dupStartIndex !== -1 && dupEndIndex !== -1) {
        lines.splice(dupStartIndex, dupEndIndex - dupStartIndex + 1);
        console.log('Duplicates removed.');
    } else {
        console.log(`Could not find duplicates range. Start: ${dupStartIndex}, End: ${dupEndIndex}`);
    }

    fs.writeFileSync(filePath, lines.join('\n'), 'utf8');
    console.log('File saved.');

} catch (err) {
    console.error(err);
}

File: Okusuri_nometane/fix_style.py
import os

file_path = r'c:\Users\kiyoshi\Github_repository\search-for-medicine\Okusuri_nometane\style.css'

with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

# 1. Fix mobile layout
# Find the line with ".certificate-characters {" inside the media query (around 1553)
target_start_index = -1
for i, line in enumerate(lines):
    if i > 1500 and ".certificate-characters {" in line:
        target_start_index = i
        break

if target_start_index != -1:
    # Check if we are replacing the right thing (flex-direction: column)
    if "column" in lines[target_start_index+1]:
        new_content = [
            "    .certificate-characters {\n",
            "        flex-direction: row;\n",
            "        flex-wrap: nowrap;\n",
            "        gap: 5px;\n",
            "        justify-content: center;\n",
            "    }\n",
            "\n",
            "    .character-signature img {\n",
            "        width: 50px;\n",
            "        height: 50px;\n",
            "    }\n",
            "\n",
            "    .character-signature span {\n",
            "        font-size: 0.8rem;\n",
            "    }\n"
        ]
        # Replace 4 lines (selector, flex-direction, gap, brace)
        lines[target_start_index:target_start_index+4] = new_content
        print("Mobile layout fixed.")
    else:
        print(f"Target content mismatch at line {target_start_index+1}: {lines[target_start_index+1]}")
else:
    print("Could not find .certificate-characters block.")

# 2. Remove duplicates
# Find start of duplicate block: ".certificate-body {" around 1577 (now shifted)
dup_start_index = -1
for i, line in enumerate(lines):
    if i > 1570 and ".certificate-body {" in line:
        dup_start_index = i
        break

# Find end of duplicate block: closing brace of duplicate mobile media query
dup_end_index = -1
if dup_start_index != -1:
    for i, line in enumerate(lines):
        if i > dup_start_index and ".certificate-characters {" in line:
            if i+1 < len(lines) and "column" in lines[i+1]:
                # This is the duplicate mobile block.
                # The block ends at i+3 (closing brace of selector)
                # The media query ends at i+4 (closing brace of media query)
                dup_end_index = i + 4
                break

if dup_start_index != -1 and dup_end_index != -1:
    del lines[dup_start_index:dup_end_index+1]
    print("Duplicates removed.")
else:
    print(f"Could not find duplicates range. Start: {dup_start_index}, End: {dup_end_index}")

with open(file_path, 'w', encoding='utf-8') as f:
    f.writelines(lines)
